go-controller/pkg/ovn/policy.go

```diff
@@ -451,17 +510,68 @@ func (oc *Controller) updateACLLoggingForPolicy(np *networkPolicy, aclLogging *A
 
 func (oc *Controller) updateACLLoggingForDefaultACLs(ns string, nsInfo *namespaceInfo) error {
 	// update namespace-scoped default deny ACLs
-	denyEgressACL, _ := buildDenyACLs(ns, "", defaultDenyPortGroup(ns, egressDefaultDenySuffix),
+	denyEgressACL, _ := buildDenyACLs(ns, defaultDenyPortGroup(ns, egressDefaultDenySuffix),
 		&nsInfo.aclLogging, knet.PolicyTypeEgress)
-	denyIngressACL, _ := buildDenyACLs(ns, "", defaultDenyPortGroup(ns, ingressDefaultDenySuffix),
+	denyIngressACL, _ := buildDenyACLs(ns, defaultDenyPortGroup(ns, ingressDefaultDenySuffix),
 		&nsInfo.aclLogging, knet.PolicyTypeIngress)
 	if err := UpdateACLLogging(oc.nbClient, []*nbdb.ACL{denyIngressACL, denyEgressACL}, &nsInfo.aclLogging); err != nil {
 		return fmt.Errorf("unable to update ACL logging for namespace %s: %w", ns, err)
 	}
 	return nil
 }
 
+// update default deny ACLs for shared port groups, if the given networkPolicy is null, update all shared port group
```

Comment:
I'm a little confused here. Today we create default deny egress/ingress pgs based on namespace. I'm not sure what this function is for or why it has anything to do with shared port groups. I also dont get why np *NetworkPolicy is an argument as the default denies are on a per namespace basis, so I dont get how you would only update default deny for 1 network policy in namespace?

Reply:
yes, but note that shared_group network policies no longer go through the normal  handleLocalPodSelector() routine so its selected local Pod's logical ports will not be added to the namespace based default deny pgs. Instead, for shared_group networl policies, we already have all the selected local logical ports in the shared port group, so just add default deny ingress/egress acls into the shared portGroup when (that complicated) condition meets.

Here,  we are updating the default deny ingress/egress acls in the shared portGroup

Reply:
also, the same function is shared by two places (one is at the end of addNetworkPolicy(), where the np argument is the specific np for the given policy,  in case aclLogging changed as createNetworkPolicy wasn't protected by nsInfo.Lock and np has not been added to nsInfo yet. Another is when aclLoggging changed, which needs to add all affected shared port group acls).

Today, there would only be one sharedPortGroup per-namespace, but it can change by just determine the right sharedPortGroup name when the policy is added. So the implementation has that support in mind.